### DISCUSSION TITLE
Issue Credential Wallet-Kit Rest-API

### DISCUSSION THREAD

**Discussion by MajdT51** [walt-id/.github#51] [Help]
Hi, I'm following the rest-api docs https://docs.walt.id/v/web-wallet/getting-started/rest-apis and trying to do everything in the background without the frontend/protals/web-wallet

Setup running using docker-compose https://docs.walt.id/v/web-wallet/getting-started/local-build/docker-build/docker-compose#docker-compose (it starts also the UIs but I won't use them)

I've already configured the issuer (using default tenant)
```
{
  "issuerApiUrl": "http://Me-MacBook-Pro.local:8082/issuer-api/default",
  "issuerClientName": "Walt.id Issuer Portal",
  "issuerDid": "did:key:z6Mkv32PFBpzN2wFzJW8byXzNErhpMx1ygdheu9cGdEBXiqx",
  "issuerUiUrl": "http://Me-MacBook-Pro.local:8082",
  "wallets": {
    "walt.id": {
      "description": "walt.id web wallet",
      "id": "walt.id",
      "presentPath": "api/siop/initiatePresentation/",
      "receivePath": "api/siop/initiateIssuance/",
      "url": "http://Me-MacBook-Pro.local:8080"
    }
  }
}
```
And I want to issue the credential https://docs.walt.id/v/web-wallet/getting-started/rest-apis/credential-issuance but have a question:
 1. I see `/**issuer-api**/{tenantId}/credentials/issuance/request` the issuer api is used here but shouldn't the Wallet be the one who trigger this and ask the issuer for a credentials? 
 In the live demo the wallet is the one who request a credential and it calls /api/**wallet**/issuance/start ?
 
 2. I run the following request as described in the docs.
 ```
 curl -X 'POST' \
  'http://me-macbook-pro.local:8080/issuer-api/default/credentials/issuance/request?walletId=walt.id&isPreAuthorized=true&issuerDid=did%3Akey%3Az6Mkv32PFBpzN2wFzJW8byXzNErhpMx1ygdheu9cGdEBXiqx' \
  -H 'accept: text/plain' \
  -H 'Content-Type: application/json' \
  -d '{
    "credentials": [
        {
        "credentialData": {
            "credentialSubject": {
                "currentAddress":["1 Boulevard de la Liberté, 59800 Lille"],"dateOfBirth":"1993-04-08","familyName":"DOE","firstName":"Jane","gender":"FEMALE","id":"did:key:z6MkqeJ3FtBccXCouTLka3WJFFugvdxiEgtZNrSmPnaioNmp","nameAndFamilyNameAtBirth":"JaneXXX DOE","personalIdentifier":"0904008084H","placeOfBirth":"LILLE, FRANCE"}
        },
        "type": "VerifiableId"
        }
    ]
}'
 ```
 and get
 ```
 http://me-MacBook-Pro.local:8080/api/siop/initiateIssuance/?issuer=http%3A%2F%2FMe-MacBook-Pro.local%3A8082%2Fissuer-api%2Fdefault%2Foidc%2F&credential_type=VerifiableId&pre-authorized_code=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJjNTRlZmFkOS02YzA1LTQzNDEtODk4MC1iY2M1OWZjM2JiMjciLCJwcmUtYXV0aG9yaXplZCI6dHJ1ZX0.0okbZXzc_iRSVUancDbX67CkktjH3mGPbdgJeDV1aWQ&user_pin_required=false
 ```
 
 Question -> What should I do next to get the issued credentials in my wallet `/api/wallet/credentials/list`
 
 3. Why is isPreAuthorized in the example is true, when did the "pre" authorization happened?
 
 Thanks :)

**Comment by taminobaumann** [comment]
@MajdT51, I think, as you were already in contact with Severin, this discussion can be closed. Please let me know otherwise. 
Best
Tamino
